### PR TITLE
MD-11907: [FE] - Reusable Graph component

### PR DESCRIPTION
**Pull request by sunnyparspec21**
JIRA: https://parspec.atlassian.net/browse/MD-11907

### CODE REVIEW

File: src/components/BarChart/BarChart.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    const renderBars = useMemo(() => barsArr.map(({ name, color, stackId, barSize }) => <Bar key={name} dataKey={name} fill={color} stackId={stackId} barSize={barSize} />), [barsArr]);
+
+    const renderReferenceLines = useMemo(
+        () =>
+            referenceLineArr?.map(({ value, color }) =>
+                orientation === 'vertical' ? (
+                    <ReferenceLine key={value} x={value} strokeWidth={2} stroke={color} label={<Label value={value} style={{ fill: color }} position="bottom" dy={3} />} />
+                ) : (
+                    <ReferenceLine key={value} y={value} strokeWidth={2} stroke={color} label={<Label value={value} style={{ fill: color }} position="left" dx={-3} />} />
+                )
```
Comment (Charul-Parspec) [inline src/components/BarChart/BarChart.tsx:34]:
```suggestion
                (
                    <ReferenceLine key={value} y={value} strokeWidth={2} stroke={color} label={<Label value={value} style={{ fill: color }} position={orientation === 'vertical' ? "bottom" : "left"} dx={orientation === 'vertical' ? 3 : -3} />} />
                )
```

File: src/components/BarChart/BarChart.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import { BarChart as RechartBarChart, Bar, XAxis, YAxis, Tooltip, ReferenceLine, Label } from 'recharts';
+
+import { Box } from '../Box';
+
+export interface BarChartProps {
+    width: number;
+    height: number;
+    data: Record<string, string | number>[];
+    orientation: 'vertical' | 'horizontal';
+    barsArr: {
```
Comment (Charul-Parspec) [inline src/components/BarChart/BarChart.tsx:11]:
```suggestion
    barsInfo: {
```

File: src/components/BarChart/BarChart.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        name: string;
+        color: string;
+        stackId?: string;
+        barSize?: number;
+    }[];
+    referenceLineArr?: {
```
Comment (Charul-Parspec) [inline src/components/BarChart/BarChart.tsx:17]:
```suggestion
    referenceLinesInfo?: {
```

**Review by Charul-Parspec — CHANGES_REQUESTED** [review]

File: src/components/BarChart/BarChart.tsx (hunk 1 above)
Comment (sunnyparspec21) [reply src/components/BarChart/BarChart.tsx:34]: @Charul-Parspec for both the line name of the prop is diffetrent for one we are using dx and for another we are using dy

Comment (Charul-Parspec) [reply src/components/BarChart/BarChart.tsx:34]: @sunnyparspec21 updated the suggestion. gist of the comment is we don't need to conditonally render ReferenceLines when props of Label can be conditionally set

Comment (sunnyparspec21) [reply src/components/BarChart/BarChart.tsx:34]: @Charul-Parspec a referenceLine component can either have x vale r y value, you can't pass both values in a single component, as you can see in one component I have passed value in x to plot the line perpendicular to x -axis and in second component I have passed value in 'y' prop o render this perpendicular to y axis. that is why we are rendering two components on condition basis.

File: src/components/BarChart/BarChart.tsx (hunk 3 above)
Comment (sunnyparspec21) [reply src/components/BarChart/BarChart.tsx]: done

File: src/components/BarChart/BarChart.tsx (hunk 2 above)
Comment (sunnyparspec21) [reply src/components/BarChart/BarChart.tsx]: done

**Review by Charul-Parspec — APPROVED** [review]

**Pull request merged**